From 695530043ca5d5654f1e236744db90ad590043cd Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sat, 18 May 2024 18:15:13 -0700
Subject: [PATCH 3/3] OpenBLAS and BLIS library search improvements

- OpenBLAS and BLIS can now be found from
  rocm_sdk build by rocm sdk builder
- selection whether to build client apps with
  OpenBLAS or BLIS can be done with
  cmake parameters

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 clients/CMakeLists.txt | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/clients/CMakeLists.txt b/clients/CMakeLists.txt
index dc8040ea..704414b5 100755
--- a/clients/CMakeLists.txt
+++ b/clients/CMakeLists.txt
@@ -122,6 +122,12 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
       elseif(EXISTS      "/opt/AMD/aocl/aocl-linux-aocc-4.0/lib_ILP64/libblis-mt.a" )
         set( BLAS_LIBRARY /opt/AMD/aocl/aocl-linux-aocc-4.0/lib_ILP64/libblis-mt.a )
         set( BLIS_INCLUDE_DIR /opt/AMD/aocl/aocl-linux-aocc-4.0/include_ILP64/ )
+      elseif(EXISTS      "${BLIS_DIR}/lib64/libblis.a")
+        set( BLAS_LIBRARY ${BLIS_DIR}/lib64/libblis.a )
+        set( BLIS_INCLUDE_DIR ${BLIS_DIR}/include/blis )
+      elseif(EXISTS      "${BLIS_DIR}/lib/libblis.a")
+        set( BLAS_LIBRARY ${BLIS_DIR}/lib/libblis.a )
+        set( BLIS_INCLUDE_DIR ${BLIS_DIR}/include/blis )
       elseif(EXISTS      "${BUILD_DIR}/deps/blis/lib/libblis.a")
         set( BLAS_LIBRARY ${BUILD_DIR}/deps/blis/lib/libblis.a )
         set( BLIS_INCLUDE_DIR ${BUILD_DIR}/deps/blis/include/blis )
@@ -134,7 +140,16 @@ if( BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_TESTS)
         message(STATUS "Linking BLIS LIB: ${BLAS_LIBRARY}")
       endif()
     else()
-      set( BLAS_LIBRARY "blas" )
+      find_package( OpenBLAS CONFIG REQUIRED )
+      message(STATUS "OpenBLAS_FOUND: ${OpenBLAS_FOUND}")
+      message(STATUS "OpenBLAS_VERSION: ${OpenBLAS_VERSION}")
+      message(STATUS "OpenBLAS_INCLUDE_DIR: ${OpenBLAS_INCLUDE_DIR}")
+      message(STATUS "OpenBLAS_INCLUDE_DIRS: ${OpenBLAS_INCLUDE_DIRS}")
+      message(STATUS "OpenBLAS_LIBRARY: ${OpenBLAS_LIBRARY}")
+      message(STATUS "OpenBLAS_LIBRARIES: ${OpenBLAS_LIBRARIES}")
+      #set( BLAS_LIBRARY ${OpenBLAS_LIBRARY} )
+      set( BLAS_LIBRARY OpenBLAS::OpenBLAS )
+      set( BLAS_INCLUDE_DIR ${OpenBLAS_INCLUDE_DIRS} )
     endif()
   else() # WIN32
     set( BLAS_INCLUDE_DIR ${OPENBLAS_DIR}/include CACHE PATH "OpenBLAS library include path" )
-- 
2.41.0

